packages/bootleg-tokens/contracts/ISharedRoyaltyTokenEnumerable.sol

```diff
@@ -0,0 +1,58 @@
+pragma solidity ^0.5.3;
+
+/**
+* @title Shared royalty NFT enumerable interface for franchisors, payments, and tokens
+*/
+contract ISharedRoyaltyTokenEnumerable {
+
+  /**
+  * @notice Gets the total number of franchisors given a token
+  * @param _token The identifier of the NFT
+  * @return uint256 The number of franchisors of that token
+  */
+  function tokenFranchisorsTotal(uint256 _token) public view returns (uint256 count);
+
+  /**
+  * @notice Enumerates the token franchisors for a given token
+  * @dev Throws if `_index` >= `tokenFranchisorsTotal(_token)` or if
+  *   `_token` is the zero address, representing an invalid NFT.
+  * @param _index The index of the franchisor to return (should be less than total franchisors for the token)
+  * @param _token The identifier of the NFT
+  * @return address The address of the franchisor
+  */
+  function tokenFranchisorsByIndex(uint256 _index, uint256 _token) public view returns (address franchisor);
+
+  /**
+  * @notice Gets the total number of payments given a token
+  * @param _token The identifier of the NFT
+  * @return uint256 The number of payments made for the token
+  */
+  function tokenPaymentsTotal(uint256 _token) public view returns (uint256 count);
+
+  /**
+  * @notice Enumerates the token payments for a given token
+  * @dev Throws if `_index` >= `tokenPaymentsTotal(_token)` or if
+  *   `_token` is the zero address, representing invalid NFTs.
+  * @param _index The index of the payment to return (should be less than total payments for the token)
+  * @param _token The identifier of the NFT
+  * @return unint256 The payment amount
+  */
+  function tokenPaymentsByIndex(uint256 _index, uint256 _token) public view returns (uint256 amount);
+
+  /**
+  * @notice Gets the number of tokens owned by a given franchisor
+  * @param _franchisor The address of the franchisor
+  * @return uint256 The number of tokens franchised
+  */
+  function tokensOfFranchisorTotal(address _franchisor) public view returns (uint256 count);
```

Comment:
This is a dupe of `franchisorBalanceOf` https://github.com/ConsenSys/web3studio-bootleg/blob/master/packages/bootleg-tokens/contracts/ISharedRoyaltyToken.sol#L16 

yeah?

Reply:
uh... 🤔 I guess the comment block is not clear. It returns the count/number of tokens owned for a franchisor, not their balance.

Reply:
You had requested that `Be able to iterate all of the tokens that a user is a franchisor of.` I thought that would require some function to produce the count, and another to provide access via index (while index < count). no?

Reply:
We discussed in person, doc'ing for reference.

`franchisorBalanceOf` isn't a great name for being able to get the total number of tokens a franchisor has ever owned. But it was done that way to align with the `721` `balanceOf`. It's made particularly more confusing by `paymentsBalanceOf` which is also in our SRT. 

We decided to keep the `franchisorBalanceOf` for consistancy, rather than having both or `tokensOfFranchisorTotal`